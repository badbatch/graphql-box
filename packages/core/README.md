# @graphql-box/core

The GraphQL Box core module.

[![npm version](https://badge.fury.io/js/%40graphql-box%2Fcore.svg)](https://badge.fury.io/js/%40graphql-box%2Fcore)
[![License: MIT](https://img.shields.io/badge/License-MIT-yellow.svg)](LICENSE)

## Installation

```bash
yarn add @graphql-box/core
```

## Usage

Read the GraphQL Box [usage guide](../../README.md#usage).

## Documentation

Read the module API [documentation](docs/README.md).
