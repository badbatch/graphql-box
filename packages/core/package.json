{
  "name": "@graphql-box/core",
  "version": "4.1.6",
  "description": "The GraphQL Box core module.",
  "author": "Dylan Aubrey <dylanaubrey@gmail.com>",
  "license": "MIT",
  "homepage": "https://github.com/badbatch/graphql-box",
  "bugs": {
    "url": "https://github.com/badbatch/graphql-box/issues"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/badbatch/graphql-box",
    "directory": "packages/core"
  },
  "publishConfig": {
    "access": "public"
  },
  "main": "lib/main/index.js",
  "module": "lib/module/index.mjs",
  "browser": "lib/browser/index.js",
  "types": "lib/types/index.d.ts",
  "keywords": [
    "graphql-box",
    "graphql",
    "client",
    "server",
    "isomorphic"
  ],
  "scripts": {
    "compile": "concurrently npm:compile:*",
    "compile-watch": "concurrently npm:compile-watch:*",
    "compile-watch:main": "yarn run compile:main --watch",
    "compile-watch:types": "yarn run compile:types --watch",
    "compile:browser": "cross-env BABEL_ENV=browser rollup -c ../../rollup.config.js",
    "compile:main": "cross-env BABEL_ENV=main babel ./src --out-dir ./lib/main --extensions \".ts\" --source-maps --config-file ../../babel.config.js",
    "compile:module": "cross-env BABEL_ENV=module babel ./src --out-dir ./lib/module --extensions \".ts\" --out-file-extension \".mjs\" --source-maps --config-file ../../babel.config.js",
    "compile:types": "tsc --build",
    "watch": "cross-env BABEL_ENV=browser rollup -c ../../rollup.config.js -w"
  },
  "dependencies": {
    "serialize-error": "8.1.0"
  },
  "peerDependencies": {
    "@babel/runtime": "< 8",
    "core-js": "< 4",
    "lodash": "< 5"
  }
}