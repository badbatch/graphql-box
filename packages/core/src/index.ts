import * as coreDefs from "./defs";
import Core from "./main";

export {
  Core as default,
  coreDefs,
};
