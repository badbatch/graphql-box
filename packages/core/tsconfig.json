{
  "extends": "../tsconfig.settings.json",
  "compilerOptions": {
    "outDir": "lib/types",
    "rootDir": "src"
  },
  "include": [
    "src"
  ]
}