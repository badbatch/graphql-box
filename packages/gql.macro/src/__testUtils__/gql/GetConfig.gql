query GetConfig {
  config {
    images {
      baseUrl
      secureBaseUrl
      backdropSizes
      logoSizes
      posterSizes
      profileSizes
      stillSizes
    }
  }
}
