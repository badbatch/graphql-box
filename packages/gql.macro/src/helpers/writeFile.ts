import { writeFileSync } from "fs";

export default (requestWhitelistPath: string, whitelist: string[]) => {
  try {
    const existingWhitelist: string[] = require(requestWhitelistPath);
    writeFileSync(requestWhitelistPath, JSON.stringify([...new Set([...existingWhitelist, ...whitelist])], null, 2));
  } catch {
    writeFileSync(requestWhitelistPath, JSON.stringify([...new Set(whitelist)], null, 2));
  }
};
