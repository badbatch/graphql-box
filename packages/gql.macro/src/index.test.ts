import { transform } from "@babel/core";
import writeFile from "./helpers/writeFile";

jest.mock("./helpers/writeFile");

const transformSourceFile = (sourceFile: string, macroConfig: Record<string, any> = {}) => {
  const transformed = transform(sourceFile, {
    filename: "test.js",
    plugins: [
      [
        "macros",
        {
          gql: {
            ...macroConfig,
            basePath: __dirname,
          },
        },
      ],
    ],
  })?.code;

  return transformed;
};

describe("gql.macro", () => {
  describe("when the macro is a tag call", () => {
    it("should transform the macro reference correctly", () => {
      const sourceFile = `
        import gql from "./packages/gql.macro/src/macro";

        const GET_CONFIG = gql\`./__testUtils__/gql/queries/GetConfig.gql\`;
        export default GET_CONFIG;
      `;

      expect(transformSourceFile(sourceFile)).toMatchSnapshot();
      expect(writeFile).toHaveBeenCalledTimes(1);
    });
  });
});
