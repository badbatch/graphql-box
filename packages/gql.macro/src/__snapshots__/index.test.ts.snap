// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gql.macro when the macro is a tag call should transform the macro reference correctly 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.default = void 0;
const GET_CONFIG = \\"query GetConfig {\\\\n  config {\\\\n    images {\\\\n      baseUrl\\\\n      secureBaseUrl\\\\n      backdropSizes\\\\n      logoSizes\\\\n      posterSizes\\\\n      profileSizes\\\\n      stillSizes\\\\n    }\\\\n  }\\\\n}\\\\n\\";
var _default = GET_CONFIG;
exports.default = _default;"
`;
